File: Week_1_Day_4.js
// Reverse an array without creating a new array.

function reverse(arr){
    for(var i=0; i < arr.length/2; i++){
        var original = arr[i];
        arr[i] = arr[arr.length-1-i]
        arr[arr.length-1-i] = original
    }
    return arr
}

console.log(reverse(["a", "b", "c", "d", "e"]))

File: Week_1_Day_4.py
# reverse an array in place

def reverse(arr):
    if len(arr) < 1:
        return []
    if len(arr) == 1:
        return arr

    for i in range(int(len(arr)/2)):
        temp = arr[i]
        arr[i] = arr[len(arr)-1-i]
        arr[len(arr)-1-i] = temp
    
    return arr

print(reverse(["a", "b", "c", "d", "e"]))